src/platform/api/pythonApi.ts

```diff
@@ -324,14 +325,18 @@ export class InterpreterService implements IInterpreterService {
         return this.didChangeInterpreters.event;
     }
 
-    @captureTelemetry(Telemetry.InterpreterListingPerf)
     @traceDecoratorVerbose('Get Interpreters', TraceOptions.Arguments | TraceOptions.BeforeCall)
     public getInterpreters(resource?: Uri): Promise<PythonEnvironment[]> {
         this.hookupOnDidChangeInterpreterEvent();
         // Cache result as it only changes when the interpreter list changes or we add more workspace folders
+        const firstTime = !!this.interpretersFetchedOnceBefore;
+        this.interpretersFetchedOnceBefore = true;
         if (!this.interpreterListCachePromise) {
             this.interpreterListCachePromise = this.getInterpretersImpl(resource);
         }
+        sendTelemetryWhenDone(Telemetry.InterpreterListingPerf, this.interpreterListCachePromise, undefined, {
+            firstTime
```

Comment:
This telemetry was lost in the refactoring, we need a way to check how long it takes to fetch the list of interpreters for the first time in the session.